### PR TITLE
[core][ci] Fix test_fault_tolerance_nested_actors_failed

### LINKED ISSUE #33738: [Tests] Fix two skipped Windows test for test_task_event_2.py
### What happened + What you expected to happen

See details.

https://github.com/ray-project/ray/pull/33704#issuecomment-1484622830

It seems to be a real bug. We should fix it. 

### Versions / Dependencies

master

### Reproduction script

n/a

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
Closes https://github.com/ray-project/ray/issues/33738

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_task_events_2.py [hunk 1]
```diff
@@ -312,12 +312,11 @@ def fail_parent(self):
         task_sleep_child.remote()
         raise ValueError("expected to fail.")
 
-    def child_actor(self):
+    def child_actor(self, wait_fn):
         a = ChildActor.remote()
-        try:
-            ray.get(a.children.remote(), timeout=2)
```
Comment (rickyyx) [inline python/ray/tests/test_task_events_2.py:318]: this timeout might not be reliable.

**Comment by rickyyx** [comment]
Window test passing: https://buildkite.com/ray-project/oss-ci-build-pr/builds/17562#0187534c-b81e-4a87-a39f-2fe3eaa65257

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
Is this also fixing `test_task_logs_info_basic`?  We should close the issue after we verify this is fixed.

**Comment by rickyyx** [comment]
> Is this also fixing `test_task_logs_info_basic`? We should close the issue after we verify this is fixed.

That was skipped now since we merged this: https://github.com/ray-project/ray/pull/34056

**Comment by rkooo567** [comment]
Gotcha. let me keep the issue open until we reenable that.

**Pull request merged**